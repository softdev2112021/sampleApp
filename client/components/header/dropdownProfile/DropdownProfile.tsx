import {
  Dropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem,
} from "reactstrap";
import { useState } from 'react';

interface DropdownProfileProps {
  userName: string;
  avatar: string;
  onLogout: () => void;
}

const DropdownProfile = (props: DropdownProfileProps) => {
  const { userName, avatar, onLogout } = props;
  const [dropdownOpen, setDropdownOpen] = useState(false);

  const toggleDropdown = () => {
    setDropdownOpen((dropdownOpen) => !dropdownOpen);
  };

  return (
    <Dropdown
      isOpen={dropdownOpen}
      toggle={toggleDropdown}
      className="dropdown navbar-user"
      tag="li"
    >
      <DropdownToggle tag="a">
        <img src={avatar} alt="profile" />
        <span className="d-none d-md-inline">{`${userName} `}</span>
        <b className="caret"></b>
      </DropdownToggle>
      <DropdownMenu className="dropdown-menu dropdown-menu-right" tag="ul">
        <DropdownItem onClick={onLogout}>Log Out</DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
};

export default DropdownProfile;
