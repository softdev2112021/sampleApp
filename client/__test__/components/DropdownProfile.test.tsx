import React from 'react';
import renderer from 'react-test-renderer';
import DropdownProfile from '../../components/header/dropdownProfile/DropdownProfile';

it('renders correctly', () => {
  const dropdownProfileProps = {
    userName: 'username1',
    avatar: 'avatar',
    onLogout: () => {},
  }

  const tree = renderer
    .create(<DropdownProfile {...dropdownProfileProps}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});