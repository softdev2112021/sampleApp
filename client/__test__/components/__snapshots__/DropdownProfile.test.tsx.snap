// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="dropdown navbar-user dropdown"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className=""
    onClick={[Function]}
  >
    <img
      alt="profile"
      src="avatar"
    />
    <span
      className="d-none d-md-inline"
    >
      username1 
    </span>
    <b
      className="caret"
    />
  </a>
  <ul
    aria-hidden={true}
    className="dropdown-menu dropdown-menu-right dropdown-menu"
    role="menu"
    tabIndex="-1"
  >
    <button
      className="dropdown-item"
      onClick={[Function]}
      role="menuitem"
      tabIndex="0"
      type="button"
    >
      Log Out
    </button>
  </ul>
</li>
`;
